scripts/konnect-funcs/supportedtypes.go

```diff
@@ -25,62 +27,94 @@ var supportedTypes = []supportedTypesT{
 			{
 				Type:              "KongKey",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneAndKeySetRef",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongKeySet",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneRef",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongCredentialBasicAuth",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneAndConsumerRefs",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongCredentialAPIKey",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneAndConsumerRefs",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongCredentialJWT",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneAndConsumerRefs",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongCredentialACL",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneAndConsumerRefs",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongCACertificate",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneRef",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongCertificate",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneRef",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongPluginBinding",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneRef",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongService",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneRef",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongRoute",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneAndServiceRefs",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongUpstream",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneRef",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongTarget",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneAndUpstreamRefs",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongVault",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneRef",
+				HasKonnectStatus:  true,
 			},
 			{
 				Type:              "KongSNI",
 				KonnectStatusType: "konnectv1alpha1.KonnectEntityStatusWithControlPlaneAndCertificateRefs",
+				HasKonnectStatus:  true,
+			},
+		},
+	},
+}
+
+var supportedTypesStandalone = []supportedTypesT{
+	{
+		Package: "v1alpha1",
+		Types: []templateDataT{
+			{
+				Type:              "KonnectGatewayControlPlane",
+				KonnectStatusType: "*KonnectEntityStatus",
+				HasKonnectStatus:  true,
+			},
+			{
+				Type:              "KonnectAPIAuthConfiguration",
+				KonnectStatusType: "*KonnectEntityStatus",
 			},
```

Comment:
What's the difference between an entity type with `HasKonnectStatus` set to `false` and `true` while having a `KonnectStatusType` defined? 🤔 Couldn't we rely purely on `KonnectStatusType` not being empty to determine whether we should generate the `GetKonnectStatus()` method?

Reply:
That makes sense, thanks for pointing this out. a8929c45b81df49e0882e3ba733eca50394cd7b9 should solve this.